docs/public/debugging-using-devfile.adoc

```diff
@@ -49,18 +31,76 @@ commands:
+
+. Download the example application:
++
+[source,sh]
+----
+$ odo create nodejs --starter
+Validation
+ ✓  Checking devfile existence [11498ns]
+ ✓  Checking devfile compatibility [15714ns]
+ ✓  Creating a devfile component from registry: DefaultDevfileRegistry [17565ns]
+ ✓  Validating devfile component [113876ns]
+
+Starter Project
+ ✓  Downloading starter project nodejs-starter from https://github.com/odo-devfiles/nodejs-ex.git [428ms]
+
+Please use `odo push` command to create the component with source deployed
+----
 
-- Now we need to create the component using `odo create nodejs`
-- Next we enable remote debugging for the component using `odo push --debug`. We can also use a custom step as the debugrun step using `odo push --debug --debug-command="custom-step"`
-- Next we port forward a local port for debugging using `odo debug port-forward`. The default local port used for debugging is 5858. If 5858 is occupied, odo will automatically pick up a local port. We can also specify the local port using, `odo debug port-forward --local-port 5858`
-- Next we need to attach the debugger to the local port. Here's a guide to do it for VS Code : https://code.visualstudio.com/docs/nodejs/nodejs-debugging#_remote-debugging[Remote Debugging]
+. Push with the `--debug` flag which is required for all debugging deployments:
++
+[source,sh]
+----
+$ odo push --debug
 
-### Check if a debugging session is running
+Validation
+ ✓  Validating the devfile [29916ns]
 
-We can check if a debugging session is running for a component by using `odo debug info`
+Creating Kubernetes resources for component nodejs
+ ✓  Waiting for component to start [38ms]
 
-```
-odo debug info
+Applying URL changes
+ ✓  URLs are synced with the cluster, no changes are required.
+
+Syncing to component nodejs
+ ✓  Checking file changes for pushing [1ms]
+ ✓  Syncing files to the component [778ms]
+
+Executing devfile commands for component nodejs
+ ✓  Executing install command "npm install" [2s]
+ ✓  Executing debug command "npm run debug" [1s]
+
+Pushing devfile component nodejs
+ ✓  Changes successfully pushed to component
+----
++
+NOTE: A custom debug command may be chosen via the `--debug-command="custom-step"` flag.
+
+. Port forward to the local port in order to access the debugging interface:
++
+[source,sh]
+----
+$ odo debug port-forward
+Started port forwarding at ports - 5858:5858
+----
++
+NOTE: A specific port may be specified using the `--local-port` flag
+
+. Checking that the debug session is running in a separate terminal window:
++
+[source,sh]
+----
+$ odo debug info
 Debug is running for the component on the local port : 5858
-```
+----
+
+. Accessing the debugger:
++
+Accessing the debugger is accessible through an assortment of tools. An example of setting up a debug interface would be through link:https://code.visualstudio.com/docs/nodejs/nodejs-debugging#_remote-debugging[VSCode's debugging interface].
```

Comment:
Since we mentioned VSCode, can we give a small example on how to attach it to the Node debugger in this case with valid snippet? For example:
```
{
  "type": "node",
  "request": "attach",
  "name": "Attach to remote",
  "address": "TCP/IP address of process to be debugged",
  "port": 9229
}
```